actions/delete_uplink.py

```diff
@@ -0,0 +1,94 @@
+import logging
+
+from flask import json
+from requests.auth import HTTPBasicAuth
+import requests
+
+
+def delete_uplink(api_auth, parameters, contexts):
+    """
+    :param api_auth: steelconnect api object
+    :type api_auth: SteelConnectAPI
+    :param parameters: json parameters from Dialogflow intent
+    :type parameters: json object
+    :return: Returns a response to be read out to user
+    :rtype: string
+    """
+    try:
+        city = parameters["City"]
+        wan_name = parameters["Wans"]
+
+    except KeyError as e:
+        error_string = "Error processing deleteUplink intent. {0}".format(e)
+        logging.error(error_string)
+        return error_string
+		
+	# Get all the sites and check whether there is a site match given city
+    data_sites = api_auth.site.list_sites().json()
+    sites = []
+    ids = []
+
+    for item in data_sites["items"]:
+        if (city.lower() == item["city"].lower()):
+            ids.append(item["id"])
+            sites.append("{}".format(item["name"]))
+			
+	# Error if no sites were found in that city
+    if (len(sites) < 1):
+        speech = "Error: No site could be found in that city"
+        return speech
+
+    # Get all the wans and check whether there is a wan match target wan user want the uplink to be deleted
```

Comment:
This comment gets a bit weird by the end. What is it trying to say?

Reply:
Ah, this is what happens when you get lazy with comments and try to make an existing one work for a new thing with minor changes, over multiple iterations. I'll go through and clarify some of these